### ISSUE TITLE
Object of type 'property' has no attribute ClientV2

### ISSUE THREAD

**Issue by rsherwood-faa** [astral-sh/ty#2110]
### Summary

```py
from acme import client

class ServerAccount:

    @property
    def client(self) -> client.ClientV2:
        if not self._client:
            self._client = client.ClientV2(self.directory, net=self._client_net)
        
        return self._client
```

ty reports
```
Object of type `property` has no attribute `ClientV2`
```

I am not sure if this is related to https://github.com/astral-sh/ty/issues/312, The code in that bug seems similar, but it looks like that issue has been closed, 

I'm using ty 0.0.3 via the ty extension for Visual Studio Code (2025.74.0). It looks like 0.14.10 is the latest. Can I update ty outside of the extension to get the latest version inside the extension?

### Version

I'm using ty 0.0.3 via the ty extension for Visual Studio Code (2025.74.0).

**Comment by rsherwood-faa** [comment]
Actually, I'm a little confused. The patch for https://github.com/astral-sh/ty/issues/312 Seems to have been addressed in https://github.com/astral-sh/ruff/pull/18121 , which was a PR against ruff. I feel like I'm missing something...

**Comment by AlexWaygood** [comment]
Sorry for the confusion. We share a lot of our foundational crates (parser, AST, several diagnostics-related crates) with Ruff, so ty's source code is all found in the Ruff repository. It makes it much easier for development. All ty issues should be reported on this tracker, though; you're in the right place :-)

**Comment by rsherwood-faa** [comment]
No errors reported for this code:
```py
import acme.client

class ServerAccount:
    @property
    def client(self) -> acme.client.ClientV2:
        if not self._client:
            self._client = acme.client.ClientV2(self.directory, net=self._client_net)
        
        return self._client
```

**Comment by rsherwood-faa** [comment]
I also discovered that the extension will use a version of ty inside a uv environment if you install it, and will fall back to the bundled version, so I can confirm that the error appeared in ty 0.4.0 and the updated code passes typechecking in 0.4.0

**Comment by sharkdp** [comment]
The problem here is that `client` is a property on `ServerAccount`, but you want to use the imported `client` in the type hint for `_client`. See https://github.com/astral-sh/ty/issues/1747 for a related discussion. The short version is that you'll probably want to rename that import (e.g. `from acme import client as acme_client`).

**Comment by rsherwood-faa** [comment]
🤯 
That makes sense, and thanks for sharing the link to the discussion. Hopefully this will help others with the same problem.
